cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreServiceUtils.java

```diff
@@ -530,4 +540,39 @@ public static boolean isSparkEngine(HiveConf hiveConf) {
     String engine = hiveConf.get("hive.execution.engine");
     return "spark".equalsIgnoreCase(engine);
   }
+
+  // This method is used to determine if Tez is enabled based on TEZ_HOME environment variable.
+  // Master prepares the explore container by adding jars available in TEZ_HOME.
+  // However this environment variable is not available to explore container itself(BaseHiveService).
+  // There we check the existence of tez using hive.execution.engine config variable.
+  private static boolean isTezAvailable() {
+    return System.getenv(Constants.TEZ_HOME) != null;
+  }
+
+  private static Set<File> getTezJars() {
+    String tezHome = System.getenv(Constants.TEZ_HOME);
+    Path tezHomeDir = Paths.get(tezHome);
+    final PathMatcher pathMatcher = tezHomeDir.getFileSystem().getPathMatcher("glob:*.jar");
+    final Set<File> tezJars = new HashSet<>();
+    try {
+      Files.walkFileTree(tezHomeDir, new SimpleFileVisitor<Path>() {
+        @Override
+        public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+          if (attrs.isRegularFile() && pathMatcher.matches(file.getFileName())) {
+            tezJars.add(file.toFile());
+          }
+          return FileVisitResult.CONTINUE;
+        }
+
+        @Override
+        public FileVisitResult visitFileFailed(Path file, IOException exc) throws IOException {
+          // Ignore error
+          return FileVisitResult.CONTINUE;
+        }
+      });
+    } catch (IOException e) {
+      LOG.warn("Exception raised while inspecting {}", tezHomeDir, e);
+    }
+    return tezJars;
```

Comment:
What happens when an exception gets thrown while walking the Tez home path? If its only one jar that we need then we can assert the size of this variable.